src/statusItem.ts

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2021 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from 'vscode';
+
+/**
+ * A {@link vscode.StatusBarItem StatusBarItem} to display the status
+ * of tasks run by this extension.
+ */
+export class StatusItem {
+
+    private item: vscode.StatusBarItem;
+
+    constructor() {
+        this.item = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+        this.item.command = `terminal.focus`;
+    }
+
+    show(message: string, addProgressIndicator = true) {
```

Comment:
If you have multiple tasks running at the same time, the first task that finishes will hide the `statusItem`. 

Should we keep a record of tasks that have shown the status item in an array and then in `hide` remove items from array and only when the array is empty hide the status item.

Reply:
I've added this, although it's not that useful for now. I've tested it by disabling the duplicate resolve/update checks and triggering some resolved while an update was running.